libretroshare/src/pqi/authssl.cc

```diff
@@ -989,411 +1002,353 @@ X509 *AuthSSLimpl::SignX509ReqWithGPG(X509_REQ *req, long /*days*/)
+			goto err;
 		}
 
 		switch(signature_info.public_key_algorithm)
 		{
-		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_RSA_ES :
-		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_RSA_S  : sigtypestring = "RSA" ;
+		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_RSA_ES: /*fallthrough*/
+		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_RSA_S:
+			sigtypestring = "RSA";
 			break ;
-
-		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_DSA    : sigtypestring = "DSA" ;
+		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_DSA:
+			sigtypestring = "DSA";
 			break ;
-
-		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_RSA_E  :
-		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_UNKNOWN:
+		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_RSA_E: /*fallthrough*/
+		case PGP_PACKET_TAG_PUBLIC_KEY_ALGORITHM_UNKNOWN: /*fallthrough*/
 		default:
 			diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_HASH_ALGORITHM_NOT_ACCEPTED ;
 			goto err ;
 		}
 
 		switch(signature_info.hash_algorithm)
 		{
-		case  PGP_PACKET_TAG_HASH_ALGORITHM_SHA1  : sigtypestring += "+SHA1"   ; break;
-		case  PGP_PACKET_TAG_HASH_ALGORITHM_SHA256: sigtypestring += "+SHA256" ; break;
-		case  PGP_PACKET_TAG_HASH_ALGORITHM_SHA512: sigtypestring += "+SHA512" ; break;
-
-			// We dont accept signatures with unknown or week hash algorithms.
-
-		case  PGP_PACKET_TAG_HASH_ALGORITHM_MD5:
-		case  PGP_PACKET_TAG_HASH_ALGORITHM_UNKNOWN:
+		case PGP_PACKET_TAG_HASH_ALGORITHM_SHA1:
+			sigtypestring += "+SHA1";
+			break;
+		case PGP_PACKET_TAG_HASH_ALGORITHM_SHA256:
+			sigtypestring += "+SHA256";
+			break;
+		case PGP_PACKET_TAG_HASH_ALGORITHM_SHA512:
+			sigtypestring += "+SHA512";
+			break;
+		// We dont accept signatures with unknown or weak hash algorithms.
+		case  PGP_PACKET_TAG_HASH_ALGORITHM_MD5: /*fallthrough*/
+		case  PGP_PACKET_TAG_HASH_ALGORITHM_UNKNOWN: /*fallthrough*/
 		default:
-			diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_HASH_ALGORITHM_NOT_ACCEPTED ;
-			goto err ;
+			diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_HASH_ALGORITHM_NOT_ACCEPTED;
+			goto err;
 		}
 
 		// passed, verify the signature itself
 
-		if (!AuthGPG::getAuthGPG()->VerifySignBin(signed_data, signed_data_length, signature->data, signature->length, pd.fpr))
+		if (!AuthGPG::getAuthGPG()->VerifySignBin(
+		            signed_data, signed_data_length, signature->data,
+		            static_cast<unsigned int>(signature->length), pd.fpr ))
 		{
-			diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_WRONG_SIGNATURE ;
+			diagnostic = RS_SSL_HANDSHAKE_DIAGNOSTIC_WRONG_SIGNATURE;
 			goto err;
 		}
 
-		RsPeerId peerIdstr ;
-		getX509id(x509, peerIdstr) ;
-
-		std::string fpr = pd.fpr.toStdString();
-		std::cerr << "Verified " << sigtypestring << " signature of certificate " << peerIdstr << ", Version " << std::hex << certificate_version
-		          << std::dec << " using PGP key " ;
-		for(uint32_t i=0;i<fpr.length();i+=4)
-			std::cerr << fpr.substr(i,4) << " " ;
-		std::cerr << std::endl;
+		Dbg1() << __PRETTY_FUNCTION__ << " Verified: " << sigtypestring
```

Comment:
the ostream output of fingerprint has spaces too? I dont think so. That's why I used an explicit output method

Reply:
Ok I'll specialize it so we will always pretty print fingerprints

Reply:
done